api/envoy/config/listener/v3/quic_config.proto

```diff
@@ -30,11 +30,14 @@ message QuicProtocolOptions {
   core.v3.QuicProtocolOptions quic_protocol_options = 1;
 
   // Maximum number of milliseconds that connection will be alive when there is
-  // no network activity. 300000ms if not specified.
+  // no network activity.
+  //
+  // If it is less than 1ms, Envoy will use 1ms. 300000ms if not specified.
   google.protobuf.Duration idle_timeout = 2;
 
   // Connection timeout in milliseconds before the crypto handshake is finished.
-  // 20000ms if not specified.
+  //
+  // If it is less than 5000ms, Envoy will use 5000ms. 20000ms if not specified.
```

Comment:
This seems very high for the minimum value. For instance, if the connection is between two hops in a datacenter, a much shorter timeout may be desired; taking away the ability for the user to configure a tight timeout seems unneccessary.

Reply:
To allow < 5s timeout, we need to expose another interface to allow users to adjust max network idle timeout during handshake in QUICHE, which currently is 5s, to be less than handshake timeout. These values are mostly hardwired for internet facing traffic where QUIC has the most advantage. We don't have any performance data for non-internet facing traffic for QUIC, especially intra-DC. And I'm not sure if QUIC for intra-DC is a proper use case given the low RTT and low packet loss. So IMO we should need the API simple as possible for now and we can always expand if needed.

Reply:
I see. That makes sense then.